docs/design/2021-04-21-unify-log-library.md

```diff
@@ -0,0 +1,165 @@
+# Proposal:
+
+- Author(s): [Yifan Xu](https://github.com/SabaPing)
+- Last updated: May 11, 2021
+- Discussion at: https://github.com/pingcap/tidb/pull/24181
+- Tracking issue: https://github.com/pingcap/tidb/issues/24190
+
+## Abstract
+
+There are heterogeneous logging libraries in PingCAP's golang projects. These different logging libraries affect the development efficiency and even the user configuration experience.
+
+It is necessary to unify those heterogeneous logging libraries.
+
+## Background
+
+Except for slow query logs, all other logs must satisfy the [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md).
+
+However, in practice, it was found that the format of logs is confusing:
+
+- There are few logging configuration instructions in the document. We need to enrich document especially the type and the format of logs each component would emit.
+- The configured logging parameters do not match the runtime logging, e.g. `tidb_stderr` is configured with text format, but the log is in json format.
+- The logs of some components do not meet the [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/23d4f9aed68a295b678e8bd909ee8479e3ba0bd1/text/2018-12-19-unified-log-format.md), e.g. `tiflash_cluster_manager`.
+- Duplicate logs, e.g. `pd_stderr` will emit both text and json logs with duplicate content (but with a few subtle differences in timestamps).
+
+## Proposal
+
+There must be something wrong with the engineering of these codes above, and they must be changed. The cost to change them is not small.
+
+Rationale - for long-term consideration, we should maintain code quality. The speed of output can be sacrificed in time if necessary.
```

Comment:
> The speed of output can be sacrificed in time if necessary.

I guess you're talking about the 'performance', or more precisely, 'maintainability should be with higher priority over performance when we have to consider the trade-offs'.